server/src/test/java/org/elasticsearch/env/NodeRepurposeCommandIT.java

```diff
@@ -68,29 +68,25 @@ public void testRepurpose() throws Exception {
         );
 
         logger.info("--> Repurposing node 1");
-        executeRepurposeCommandForOrdinal(noMasterNoDataSettings, indexUUID, 1);
+        executeRepurposeCommandForOrdinal(noMasterNoDataSettings, indexUUID, 1, 1);
 
         ElasticsearchException lockedException = expectThrows(ElasticsearchException.class,
-            () -> executeRepurposeCommandForOrdinal(noMasterNoDataSettings, indexUUID, 0)
+            () -> executeRepurposeCommandForOrdinal(noMasterNoDataSettings, indexUUID, 0, 1)
         );
 
         assertThat(lockedException.getMessage(), containsString(NodeRepurposeCommand.FAILED_TO_OBTAIN_NODE_LOCK_MSG));
 
         logger.info("--> Starting node after repurpose");
         internalCluster().startCoordinatingOnlyNode(Settings.EMPTY);
 
-        ensureGreen();
-
-        // docs gone
-        assertFalse(client().prepareGet(indexName, "type1", "1").get().isExists());
         assertTrue(indexExists(indexName));
 
         logger.info("--> Restarting and repurposing other node");
 
         internalCluster().stopRandomNode(s -> true);
         internalCluster().stopRandomNode(s -> true);
 
-        executeRepurposeCommandForOrdinal(noMasterNoDataSettings, indexUUID, 0);
+        executeRepurposeCommandForOrdinal(noMasterNoDataSettings, indexUUID, 0, 0);
 
         internalCluster().startMasterOnlyNode();
```

Comment:
This is not easy to follow. We've repurposed both nodes as non-master and non-data. And now we're starting them as "master" and "data" correspondingly. 
I think, that we at least need to add a comment that describes what we're doing here. 
Like "now both nodes are coordinating, let's start one of them as a master node and the second one as the data node, this does not require repurpose tool invocation. However, since they were converted to coordinating nodes previously, all data and metadata should be lost and we can verify it".